infrastructure/zk/src/config.ts

```diff
@@ -75,24 +49,47 @@ export function collectVariables(config: any, prefix: string = ''): Map<string,
     return variables;
 }
 
-export function loadConfig(env?: string, forceAll: boolean = false): any {
+export function loadConfig(env?: string): object {
     env ??= process.env.ZKSYNC_ENV!;
-    let config = {};
 
-    if (forceAll || !env.startsWith('ext-node')) {
-        for (const configFile of CONFIG_FILES) {
-            const localConfig = loadConfigFile(`etc/env/base/${configFile}`);
-            deepExtend(config, localConfig);
-        }
+    let configPath = `${process.env.ZKSYNC_HOME}/etc/env/${env}.toml`;
+    const config = {};
+    if (fs.existsSync(configPath)) {
+        loadConfigRecursive(config, configPath, []);
+    } else {
+        configPath = `${process.env.ZKSYNC_HOME}/etc/env/dev.toml`;
+        loadConfigRecursive(config, configPath, []);
     }
+    return config;
+}
 
-    const overridesPath = `${process.env.ZKSYNC_HOME}/etc/env/${env}.toml`;
-    if (fs.existsSync(overridesPath)) {
-        const overrides = loadConfigFile(overridesPath);
-        deepExtend(config, overrides);
+function loadConfigRecursive(config: object, configPath: string, calledFrom: string[]) {
```

Comment:
Another sort of unrelated change: I've found the current config compilation logic to be overly brittle, so I've devised something similar to Spring 🙃 I think it should be easy to decipher from the changed config TOMLs, but essentially `_metadata.base` is an array of paths to config files (relative to `etc/env` dir) that this config inherits. In this way, copy-paste between Docker and "ordinary" variants of configs is eliminated, and the `dev` config doesn't require special handling.

Reply:
I'd say -0 on this change. It wasn't pretty before, it is more complex now. `zk` seems to be a hot topic and point of confusion lately, so probably I'd expect a bit more discussion here.

However, given that 
a) it's already done, and
b) `zk` would likely be replaced with something else mid-term.

Still, probably it makes sense to contact @ly0va to make sure that it doesn't conflict with the work he and Kalmon are doing for hyperchains.